terraform/modules/monitoring/grafana.tf

```diff
@@ -0,0 +1,134 @@
+module "grafana" {
+  source                           = "../app"
+  execution_role_arn               = var.execution_role_arn
+  vpc_id                           = var.vpc_id
+  cluster_id                       = aws_ecs_cluster.cluster.id
+  service_name                     = "grafana"
+  subnets                          = var.private_subnets
+  mesh_name                        = var.mesh_name
+  service_discovery_namespace_id   = var.service_discovery_namespace_id
+  service_discovery_namespace_name = var.service_discovery_namespace_name
+  extra_security_groups            = [var.govuk_management_access_sg_id]
+  desired_count                    = var.desired_count
+  custom_container_services        = [{ container_service = "grafana", port = 3000, protocol = "http" }]
+
+  load_balancers = [{
+    target_group_arn = aws_lb_target_group.grafana_public.arn
+    container_name   = "grafana"
+    container_port   = 3000
+  }]
+}
+
+#
+# IAM
+#
+
+resource "aws_iam_role" "grafana_task" {
+  name        = "grafana_task_role"
+  description = "Role for GOV.UK Publishing app containers (ECS tasks) to talk to other AWS services."
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Effect": "Allow"
+    }
+  ]
+}
+EOF
+}
+
+# Proxy authorization for Grafana
+# https://docs.aws.amazon.com/app-mesh/latest/userguide/proxy-authorization.html
+resource "aws_iam_role_policy_attachment" "grafana_appmesh_envoy_access" {
+  role       = aws_iam_role.grafana_task.id
+  policy_arn = "arn:aws:iam::aws:policy/AWSAppMeshEnvoyAccess"
+}
+
+resource "aws_iam_role_policy_attachment" "grafana_cloudwatch_read_access" {
+  role       = aws_iam_role.grafana_task.id
+  policy_arn = "arn:aws:iam::aws:policy/CloudWatchReadOnlyAccess"
+}
+
+#
+# Internet-facing load balancer
+#
+
+# TODO: use a single, ACM-managed cert with both domains on. There is already
+# such a cert in integration/staging/prod (but it needs defining in Terraform).
+data "aws_acm_certificate" "public_lb_default" {
+  domain   = "*.test.govuk.digital"
+  statuses = ["ISSUED"]
+}
+
+data "aws_acm_certificate" "public_lb_alternate" {
+  domain   = "*.test.publishing.service.gov.uk"
+  statuses = ["ISSUED"]
+}
+
+resource "aws_lb" "grafana_public" {
```

Comment:
There's a `public-load-balancer` module you can make use of, which might enable you to remove the ALB resources here.

Reply:
I will have a look. thanks.

Reply:
done, I had to add some options to the `public-load-balancer` module